pyvista/utilities/fileio.py

```diff
@@ -233,3 +249,106 @@ def read_exodus(filename,
 
     reader.Update()
     return pyvista.wrap(reader.GetOutput())
+
+
+def read_meshio(filename, file_format = None):
+    """Read any mesh file using meshio."""
+    from meshio._vtk import (
+        meshio_to_vtk_type,
+        vtk_type_to_numnodes,
+    )
+    
+    # Read mesh file
+    mesh = meshio.read(filename, file_format)
+
+    # Extract cells from meshio.Mesh object
+    offset = []
+    cells = []
+    cell_type = []
+    cell_data = {}
+    next_offset = 0
+    for k, v in mesh.cells.items():
+        vtk_type = meshio_to_vtk_type[k]
+        numnodes = vtk_type_to_numnodes[vtk_type]
+        offset += [next_offset+i*(numnodes+1) for i in range(len(v))]
+        cells.append(numpy.hstack((numpy.full((len(v), 1), numnodes), v)).ravel())
+        cell_type += [vtk_type] * len(v)
+        next_offset = offset[-1] + numnodes + 1
+
+        # Extract cell data
+        if k in mesh.cell_data.keys():
+            for kk, vv in mesh.cell_data[k].items():
+                if kk in cell_data:
+                    cell_data[kk] = numpy.concatenate((cell_data[kk], vv))
+                else:
+                    cell_data[kk] = vv
+
+    # Create pyvista.UnstructuredGrid object
+    grid = pyvista.UnstructuredGrid(
+        numpy.array(offset),
+        numpy.concatenate(cells),
+        numpy.array(cell_type),
+        numpy.array(mesh.points, numpy.float64),
+    )
+
+    # Set point data
+    for k, v in mesh.point_data.items():
+        data = vtk.util.numpy_support.numpy_to_vtk(numpy.array(v, numpy.float64))
+        data.SetName(k)
+        grid.GetPointData().AddArray(data)
+
+    # Set cell data
+    for k, v in cell_data.items():
+        data = vtk.util.numpy_support.numpy_to_vtk(numpy.array(v, numpy.float64))
+        data.SetName(k)
+        grid.GetCellData().AddArray(data)
+    return grid
+
+
+def save_meshio(filename, mesh, file_format = None, **kwargs):
+    """Save mesh to file using meshio."""
+    from meshio._vtk import vtk_to_meshio_type
+
```

Comment:
Make sure relative paths like `"~/Desktop.foo.vti"` will work
```suggestion
    filename = os.path.abspath(os.path.expanduser(filename))
```